### PR TITLE
Add RingCentral recipe

### PR DESCRIPTION
**Pull request by deadmeu**
This PR adds a recipe for the RingCentral unified communications app, which is the latest version of their teams messaging product (succeeding their classic messaging product _Glip_).

Both direct and indirect message badges have been implemented. Notifications don't work (I think the cause could be a permission issue) but that is something I'll look at in the future.

I've had a [basic recipe available](https://github.com/deadmeu/ferdi-ringcentral) for a while, but have only just bothered to clean things up to be published now.

This PR may resolve future issues like getferdi/ferdi#1388.

### CODE REVIEW

File: recipes/proton-mail/webview.js [hunk 1]
```diff
@@ -2,10 +2,10 @@ module.exports = Ferdi => {
   const getMessages = () => {
     let unreadCount = 0;
     // Loop over all displayed counters and take the highest one (from the "All Mail" folder)
-    document.querySelectorAll('.navigation-counter-item').forEach(counterElement => {
+    for (const counterElement of document.querySelectorAll('.navigation-counter-item')) {
```
Comment (vraravam) [inline recipes/proton-mail/webview.js:5]: Why are you editing the `proton-mail` recipe in this PR?

File: recipes/ringcentral/README.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# RingCentral for Ferdi
+This is a Ferdi recipe for the RingCentral unified communications app (formerly Glip).
```
Comment (vraravam) [inline recipes/ringcentral/README.md:2]: a `README.md` is no longer required - please remove

**Review by vraravam — CHANGES_REQUESTED** [review]

File: recipes/proton-mail/webview.js (hunk 1 above)
Comment (deadmeu) [reply recipes/proton-mail/webview.js:5]: I think it was the pre-commit `eslint . --fix` command that made the change.

**Review by vraravam — APPROVED** [review]

**Comment by kytwb** [comment]
@all-contributors please add @deadmeu for code

**Pull request merged**